simplex/block.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/simplex"
+
+	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+)
+
+var _ simplex.BlockDeserializer = (*blockDeserializer)(nil)
+
+var _ simplex.Block = (*Block)(nil)
+
+var _ simplex.VerifiedBlock = (*Block)(nil)
+
+var errBlockAlreadyVerified = errors.New("block has already been verified")
```

Comment:
We typically use just a single var block rather than multiple when we are defining a group of things.
```suggestion
var (
	_ simplex.BlockDeserializer = (*blockDeserializer)(nil)
	_ simplex.Block             = (*Block)(nil)
	_ simplex.VerifiedBlock     = (*Block)(nil)

	errBlockAlreadyVerified = errors.New("block has already been verified")
)
```